mne_bids/tests/test_read.py

```diff
@@ -327,20 +330,64 @@ def test_handle_coords_reading():
     raw.set_montage(montage)
     write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
 
-    # read in the data and assert montage is the same
-    bids_fname = bids_basename + "_ieeg.edf"
+    # obtain the sensor positions and assert ch_coords are same
     raw_test = read_raw_bids(bids_fname, bids_root)
-
-    # obtain the sensor positions
     orig_locs = raw.info['dig'][1]
     test_locs = raw_test.info['dig'][1]
     assert orig_locs == test_locs
     assert not object_diff(raw.info['chs'], raw_test.info['chs'])
 
-    # test error message if electrodes don't match
+    # read in the data and assert montage is the same
+    # regardless of 'm', 'cm', 'mm', or 'pixel'
+    scalings = {'m': 1., 'cm': 100., 'mm': 1000.}
+    coordsystem_fname = _find_matching_sidecar(bids_fname, bids_root,
+                                               suffix='coordsystem.json')
     electrodes_fname = _find_matching_sidecar(bids_fname, bids_root,
                                               "electrodes.tsv",
                                               allow_fail=True)
+    orig_electrodes_dict = _from_tsv(electrodes_fname,
+                                     [str, float, float, float, str])
+    for coord_unit, scaling in scalings.items():
+        # update coordinate SI units
+        _update_sidecar(coordsystem_fname, 'iEEGCoordinateUnits', coord_unit)
+        electrodes_dict = _from_tsv(electrodes_fname,
+                                    [str, float, float, float, str])
+        for axis in ['x', 'y', 'z']:
+            electrodes_dict[axis] = \
+                np.multiply(orig_electrodes_dict[axis], scaling)
+        _to_tsv(electrodes_dict, electrodes_fname)
+
+        # read in raw file w/ updated montage
+        raw_test = read_raw_bids(bids_fname, bids_root)
+
+        # obtain the sensor positions and make sure they're the same
+        assert_dig_allclose(raw.info, raw_test.info)
+
+    # check that coordinate systems can be used and defaults to mri
```

Comment:
this part of the test now fails after my last commit. Please have a look. Than I think it's good enough

Reply:
In terms of setting the coordinate frame, this line in mne-python prevents this from setting it as 'unknown':

mne-python/mne/channels/montage.py:664
>`mnt_head = _get_montage_in_head(montage)`

Inside function: `mne.channels.montage._set_montage`. As a result, `raw.set_montage(montage)` now has all coordinate frames in `head` frame. Idk what's the proper fix in mne-python, but I put up a fix to the unit test for mne-bids for now. Lmk if this is okay.